verification/policy.go

```diff
@@ -203,3 +203,44 @@ func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
 	// No errors
 	return nil
 }
+
+// getApplicableTrustPolicy returns a pointer to the deep copied TrustPolicy statement that applies to the given
+// registry URI. If no applicable trust policy is found, returns an error
+// see https://github.com/notaryproject/notaryproject/blob/main/trust-store-trust-policy-specification.md#selecting-a-trust-policy-based-on-artifact-uri
+func (policyDoc *PolicyDocument) getApplicableTrustPolicy(artifactUri string) (*TrustPolicy, error) {
+
+	artifactPath, err := getArtifactPathFromUri(artifactUri)
+	if err != nil {
+		return nil, err
+	}
+
+	var wildcardPolicy *TrustPolicy
+	var applicablePolicy *TrustPolicy
+	for _, policyStatement := range policyDoc.TrustPolicies {
+		if isPresent(wildcard, policyStatement.RegistryScopes) {
+			wildcardPolicy = policyStatement.deepCopy() // we need to deep copy because we can't use the loop variable address. see https://stackoverflow.com/a/45967429
+		} else if isPresent(artifactPath, policyStatement.RegistryScopes) {
+			applicablePolicy = policyStatement.deepCopy()
+		}
+	}
+
+	if applicablePolicy != nil {
+		// a policy with exact match for registry URI takes precedence over a wildcard (*) policy.
+		return applicablePolicy, nil
+	} else if wildcardPolicy != nil {
+		return wildcardPolicy, nil
+	} else {
+		return nil, fmt.Errorf("artifact %q has no applicable trust policy", artifactPath)
```

Comment:
Nit, I'd prefer `artifactUri` in the error message which is the user provided input, `artifactPath` is calculated in our internal logic.

```suggestion
		return nil, fmt.Errorf("artifact %q has no applicable trust policy", artifactUri)
```